main/src/ca/uwaterloo/flix/language/phase/Verifier.scala

```diff
@@ -229,32 +236,92 @@ object Verifier {
 
         case AtomicOp.Tag(sym) =>
           val List(t1) = ts
+          // We do not know the case terms so t1 cannot be verified.
           check(expected = MonoType.Enum(sym.enumSym))(actual = tpe, loc)
 
         case AtomicOp.Untag(sym) =>
           val List(t1) = ts
           check(expected = MonoType.Enum(sym.enumSym))(actual = t1, loc)
+          // We do not know the case terms so tpe cannot be verified.
+          tpe
+
+        case AtomicOp.ArrayLength =>
+          val List(t1) = ts
+          check(expected = MonoType.Int32)(actual = tpe, loc)
+          checkWith(t1, { case arr: MonoType.Array => arr }, "Array", loc)
```

Comment:
@magnus-madsen Is this too much? 

Here we want t1 to be some array type but the element type is unconstrained here, so I can't construct an array to compare to. You could write
```
val MonoType.Array(_) = t1
```
But that wouldn't make a nice exception. That made me make this function. 

I also like it for other places - if t1 has to be `Array[t]` and t2 has to be `t` then the array part is really expected, but the `t` part is just that they have to agree. So its nice to split the destruction of the array and the checking of the element type

Reply:
Just wanted to skip writing one for ref/lazy/arrow/array etc. but i can do that

Reply:
How many? Its like 7 no?

Reply:
Its also a bit weird to write `getArrayElement` and ignore the result

Reply:
I see what you mean.

Reply:
assertArrayTypeAndGetElm

Reply:
IDK: Just not strings please :)